index_async.js

```diff
@@ -129,19 +123,30 @@ pepperMint(user, pass)
     getCoinUsdBalance(coin).then((coinUsdBalance) => {
       balanceCnt += 1;
       totalHoldings += coinUsdBalance;
-      console.log('Updating Mint balance for ' + coin.name + ' to: $', coinUsdBalance.toFixed(2));
-      mint.updateAccount({
-        accountId: coin.mintAcctId,
-        accountValue: coinUsdBalance
-      });
+
+      //update Mint only if user has any coin holdings
+      if(coin.addrs.length > 0){
+        console.log('$$$$$ Updating Mint balance for ' + coin.name + ' to: $', coinUsdBalance.toFixed(2));
+        mint.updateAccount({
+          accountId: coin.mintAcctId,
+          accountValue: coinUsdBalance
+        });
+      }
+
       if (balanceCnt >= coinCnt) {
         console.log('==============================================');
         console.log('totalHoldings: $' + totalHoldings.toFixed(2));
         console.log('costBasis2017: $' + costBasis2017.toFixed(2));
         console.log('costBasis2018: $' + costBasis2018.toFixed(2));
         console.log('totalCostBasis: $' + totalCostBasis.toFixed(2));
         console.log('gains: $' + (totalHoldings-totalCostBasis).toFixed(2));
-        console.log('return: ' + (totalHoldings/totalCostBasis*100).toFixed(2) + '%');
+
+        let returns = 0
+        if(totalCostBasis > 0){
```

Comment:
what's your thought around re-factoring and re-formatting this code?

though it doesn't hurt to ensure that `totalCostBasis` is positive, there is no legitimate scenario where it would be negative, since it is meant to represent the costs initially paid for the assets (and to later compare current holdings with)